src/components/tree-views-container.js

```diff
@@ -495,16 +496,17 @@ const TreeViewsContainer = () => {
         return () => {
             wsToClose.close();
         };
-    }, []);
+    }, [dispatch]);
 
     const onUpdateStudies = useCallback(
         (event) => {
             console.debug('Received Update Studies notification', event);
+
             let eventData = JSON.parse(event.data);
             if (!eventData.headers) {
                 return;
             }
-
+            dispatch(directoryUpdated(eventData));
```

Comment:
Maybe this is the only thing to do when initializing the websocket : 
```
wsRef.current.onmessage = (event) => { 
  let eventData = JSON.parse(event.data);
  if (!eventData.headers) {
      return;
  }
  dispatch(directoryUpdated(eventData));
};
```
Then the current onUpdateStudies function becomes an effect which proc when directoryUpdated changes in the store and consume differents types of notifications, you know like it's done in GridStudyApp